core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -429,7 +486,7 @@ class LogSegment private[log] (val log: FileRecords,
    * The time index entry appended will be used to decide when to delete the segment.
    */
   def onBecomeInactiveSegment() {
-    maybeAppendToTimeIndex(maxTimestampSoFar, offsetOfMaxTimestamp, skipFullCheck = true)
+    timeIndex.maybeAppend(maxTimestampSoFar, offsetOfMaxTimestamp, skipFullCheck = true)
```

Comment:
Hmm, I seem to have missed looking into this call in more detail. Do you think we need to handle a potential offset overflow when `onBecomeInactiveSegment` is called from `roll()`?

Reply:
Double check to be sure, but I think we already protect updates to `offsetOfMaxTimestamp`. This field seems is updated in three cases 1) append, 2) recovery, and 3) loading from the index. We do validation in the first two cases and, in the third, the offset should already be in the right range (since it was already appended to the index).